### PR TITLE
Componentize app with Minidux

### PR DESCRIPTION
**Pull request by laurengarcia**
So I found a great article that neatly explains why Minidux is a great tool for componentizing the app (Minidux is a pared down version of Redux):
https://css-tricks.com/learning-react-redux/

The basic idea is that we want a single source of truth for our app state that the components can dynamically subscribe to, rather than all the chaos of one component notifying another that something happened, and then the 2nd notifying a 3rd, and so and so on as commonly happens as browser apps complexify. Do read that article^^. Minidux is awesome and fixes this problem efficiently and effectively.

I went ahead and componentized the first few sample pieces that needed it, next week I (or someone else) can finish up so that the *only* logic in index.js (our app logic) is main() and getArchive() -- and of course the component callbacks for subscribing to state changes. Everything else in index.js should become a component IMO, whether it makes use of state or not (ResetButton does not, for example).

The pattern I've arrived at since Minidux only allows for ONE subscriber you can see at index.js, lines 17-32. Each of the components is init'd and pushed into an array. Then, when a state change dispatch to Minidux happens via store.dispatch(), the subscriber on index.js line 26 walks through each component looking for a component method on each called update(). If update() is present on the component, that means it wants to be notified of an application state change and contains internal handlers for doing so. 

Honestly, this is so lightweight and awesome for what its giving us, I am really stoked on it. If it takes a minute to wrap head around, its worth it. ++++ Props again to authors of Yo-Yo and Minidux! ++++

### CODE REVIEW

**Comment by karissa** [comment]
this is looking awesome~!

**Comment by juliangruber** [comment]
i'll try to give this a good review tomorrow (tuesday) :)

File: components/speed-display/index.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      s.archive.on('download', function (data) {
+        self._calculate('download', data)
+      })
+    }
+  }
+}
+
+SpeedDisplay.prototype._calculate = function (direction, data) {
+  if (!data.length) return
+  var bytesPerSecond = this._speedometer(data.length)
```
Comment (mafintosh) [inline components/speed-display/index.js]: you need a speedometer instance for both upload and download speed

Comment (laurengarcia) [inline components/speed-display/index.js]: of course, duh, thanks. been so focused on the minidux parts i neglected this.

**Pull request merged**